model/host/host.go

```diff
@@ -316,13 +316,10 @@ func (h *Host) MarkAsProvisioned() error {
 // ClearRunningTask unsets the running task key on the host and updates the last task
 // completed fields.
 func (host *Host) ClearRunningTask(prevTaskId string, finishTime time.Time) error {
-	host.LastTaskCompleted = prevTaskId
-	host.LastTaskCompletedTime = finishTime
-	host.RunningTask = ""
-	event.LogHostRunningTaskCleared(host.Id, prevTaskId)
-	return UpdateOne(
+	err := UpdateOne(
 		bson.M{
-			IdKey: host.Id,
+			IdKey:          host.Id,
+			RunningTaskKey: host.RunningTask,
```

Comment:
Why set this? Aren't we clearing a running task?

Reply:
This is the query portion of the update, in other words, don't update if the task we think is the running task is not set as the running task